cpp/include/raft/core/device_span.hpp

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/core/span.hpp>
+
+namespace raft {
+
+/**
+ * @brief A span class for device pointer.
```

Comment:
Given that `std::span` is _just_ a pointer and a length, with no accessor or other information, does it still make sense for RAFT to have a `span` class separate from `mdspan`?

Reply:
I've brought up in the past, actually, that a `span` and `mdspan` are similar enough that we could probably just specialize the mdspan into a span. I think for now, while we have users downstream still using span, we should keep it in the meantime while we continue the discussion and consolidate in a future follow-up.

Reply:
Created issue to continue the discussion more formally w/ different stakeholders: https://github.com/rapidsai/raft/issues/817